zmon_worker_monitor/builtins/plugins/datapipeline.py

```diff
@@ -0,0 +1,77 @@
+#!/usr/bin/env python
+# -*- coding: utf-8 -*-
+
+import boto3
+import fnmatch
+import logging
+import requests
+
+
+from zmon_worker_monitor.adapters.ifunctionfactory_plugin import IFunctionFactoryPlugin, propartial
+
+
+logging.getLogger('botocore').setLevel(logging.WARN)
+
+
+class DataPipelineWrapperFactory(IFunctionFactoryPlugin):
+    def __init__(self):
+        super(DataPipelineWrapperFactory, self).__init__()
+
+    def configure(self, conf):
+        return
+
+    def create(self, factory_ctx):
+        """
+        Automatically called to create the check function's object
+        :param factory_ctx: (dict) names available for Function instantiation
+        :return: an object that implements a check function
+        """
+        return propartial(DataPipelineWrapper, region=factory_ctx.get('entity').get('region', None))
+
+
+def get_region():
+    r = requests.get('http://169.254.169.254/latest/dynamic/instance-identity/document', timeout=3)
+    return r.json()['region']
+
+
+def matches(dimensions, filters):
+    for key, pattern in filters.items():
+        if not fnmatch.fnmatch(''.join(dimensions.get(key, '')), pattern):
+            return False
+    return True
+
+
+# create a dict of keys from a list of dicts
+def create_dict_from_list_of_fields(fields):
+    fields_dict = {}
+    for field in fields:
+        fields_dict[str(field['key'])] = str(field['stringValue'])
```

Comment:
without str() the strings would be 'u''someString' which I find a little annoying :D